protocol/src/main/protobuf/Types.proto

```diff
@@ -73,6 +74,7 @@ message MajorType {
   optional int32 precision = 4; // used for decimal types
   optional int32 scale = 5; // used for decimal types
   optional int32 timeZone = 6; // used by TimeStamp type
+  repeated MinorType sub_type = 7; // used by Union type
```

Comment:
We basically don't allow a non-leaf/non-root node of an expression tree to have a complex type. Expressions where an intermediate node are complex, the expression is split apart into separate operators. So we only care about the subtypes of a map or list vector when reading from the vector.

In the case of a Union vector, it is possible that intermediate nodes might have Union type. We pass the possible subtypes up the tree so that the code generated only includes the code needed for the specific types that are present, rather than having to generate code for every type at every node.